Automattic/wp-calypso
# Add the “Live Preview” option on the Themes Showcase page

Related to https://github.com/Automattic/wp-calypso/issues/79219, https://github.com/Automattic/wp-calypso/issues/79218, https://github.com/Automattic/wp-calypso/pull/79775

## Proposed Changes

Add the “Live Preview” option on the Themes Showcase page. 

| before | after |
|--------|--------|
| <img width="380" alt="Screen Shot 2023-08-01 at 16 04 41" src="https://github.com/Automattic/wp-calypso/assets/5287479/f8eba2de-3e7b-4d13-9ca4-fe65f9e70905"> | <img width="388" alt="Screen Shot 2023-08-01 at 16 03 56" src="https://github.com/Automattic/wp-calypso/assets/5287479/1ea30aaa-4336-4660-bf2d-1da6e3bb233d"> |


- This PR is a part of #79219.
- This PR focuses on adding the new button. I'll take care of other UI changes, such as changing the copy of the "Live demo".
- ⚠️ This PR should be merged after https://github.com/Automattic/wp-calypso/pull/80017.

## Testing Instructions



Simple sites

- Patch D117384-code into your sandbox.
- Go to the Theme Showcase page.
- Click three-dot on any theme.
- See if clicking the "Live Preview" option redirects you to the Site Editor (Block Theme Previews).
	- You will not see the option when a theme is;
		- your active theme
		- is not FullSiteEditing compatible
		- a theme with a static page as a homepage 
		- a 3rd party theme
		- a premium theme
		- a woocommerce theme

Atomic sites

- Go to the Theme Showcase page.
- Click three-dot on any theme.
- Currently, you must install a theme before using the Live Preview on Atomic sites. pbxlJb-3Uv-p2#known-issues
	- When you're using Calypso, activate it once and then reactivate the original theme. 
	- When you're using the classic view, just install it.
- ~~See if clicking the "Live Preview" option redirects you to the Site Editor (Block Theme Previews).~~ -> I am not yet sure how to patch it to make BTP work (c.f. https://github.com/Automattic/dotcom-forge/issues/3322). We can just check if the "Live Preview" option exists or not.
	- You will not see the button when a theme is;
		- your active theme
		- is not FullSiteEditing compatible
		- a theme with a static page as a homepage 
		- not installed (not listed in "My Themes")

Please see pbxlJb-3Uv-p2#known-issues about unsupported themes. 

## Pre-merge Checklist
